src/test/java/de/tum/in/www1/artemis/exam/ProgrammingExamIntegrationTest.java

```diff
@@ -0,0 +1,295 @@
+
+    @BeforeEach
+    void initTestCase() {
+        userUtilService.addUsers(TEST_PREFIX, NUMBER_OF_STUDENTS, NUMBER_OF_TUTORS, 0, 1);
+
+        course1 = courseUtilService.addEmptyCourse();
+        student1 = userUtilService.getUserByLogin(TEST_PREFIX + "student1");
+        exam1 = examUtilService.addExam(course1);
+
+        bitbucketRequestMockProvider.enableMockingOfRequests();
+
+        ParticipantScoreScheduleService.DEFAULT_WAITING_TIME_FOR_SCHEDULED_TASKS = 200;
+        participantScoreScheduleService.activate();
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        bitbucketRequestMockProvider.reset();
+        bambooRequestMockProvider.reset();
+        if (programmingExerciseTestService.exerciseRepo != null) {
+            programmingExerciseTestService.tearDown();
+        }
+
+        ParticipantScoreScheduleService.DEFAULT_WAITING_TIME_FOR_SCHEDULED_TASKS = 500;
+        participantScoreScheduleService.shutdown();
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testUpdateExam_rescheduleProgramming_visibleAndStartDateChanged() throws Exception {
+        // Add a programming exercise to the exam and change the dates in order to invoke a rescheduling
+        var programmingEx = programmingExerciseUtilService.addCourseExamExerciseGroupWithOneProgrammingExerciseAndTestCases();
+        var examWithProgrammingEx = programmingEx.getExerciseGroup().getExam();
+
+        ZonedDateTime visibleDate = examWithProgrammingEx.getVisibleDate();
+        ZonedDateTime startDate = examWithProgrammingEx.getStartDate();
+        ZonedDateTime endDate = examWithProgrammingEx.getEndDate();
+        examUtilService.setVisibleStartAndEndDateOfExam(examWithProgrammingEx, visibleDate.plusSeconds(1), startDate.plusSeconds(1), endDate);
+
+        request.put("/api/courses/" + examWithProgrammingEx.getCourse().getId() + "/exams", examWithProgrammingEx, HttpStatus.OK);
+        verify(instanceMessageSendService).sendProgrammingExerciseSchedule(programmingEx.getId());
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testUpdateExam_rescheduleProgramming_visibleDateChanged() throws Exception {
+        var programmingEx = programmingExerciseUtilService.addCourseExamExerciseGroupWithOneProgrammingExerciseAndTestCases();
+        var examWithProgrammingEx = programmingEx.getExerciseGroup().getExam();
+        examWithProgrammingEx.setVisibleDate(examWithProgrammingEx.getVisibleDate().plusSeconds(1));
+        request.put("/api/courses/" + examWithProgrammingEx.getCourse().getId() + "/exams", examWithProgrammingEx, HttpStatus.OK);
+        verify(instanceMessageSendService).sendProgrammingExerciseSchedule(programmingEx.getId());
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testUpdateExam_reschedule_startDateChanged() throws Exception {
```

Comment:
For consistency with other test names:
```suggestion
    void testUpdateExam_rescheduleProgramming_startDateChanged() throws Exception {
```

Reply:
done